utils/lock_dependencies_violations.py

```diff
@@ -0,0 +1,109 @@
+from typing import List, Set, Dict, Tuple
+from adjacency_matrix import AdjacencyMatrix
+from acceptance_variants import generate_acceptance_variants
+from traces_to_matrix import traces_to_adjacency_matrix
+from dependencies import ExistentialType
+
+
+def locked_dependencies_preserved(initial_matrix: AdjacencyMatrix, modified_matrix: AdjacencyMatrix, locked_dependencies: Dict[Tuple[str, str], Tuple[bool, bool]], deletion_allowed: Tuple[bool, bool]) -> bool:
+    """
+    Check if provided constraints are fullfilled, meaning that dependencies which are locked are not modified when applying the change operation 
+    
+    Args:
+        initial_matrix: The adjacency matrix before performing the change operation
+        modified_matrix: The adjacency matrix after performing the change operation
+        locked_dependencies: List of dependencies which are locked (activity1, activity2, dependency_type) which should be ensured. The first tuples describes the activities and the second one (temporal, dependency) 
+        describes which of the depencies are locked  
+        deletion_allowed: Indicates if the deletion of the depenedency by deleting one of the activities is allowed, tuples indicates whether the from activity (first element) or to activity (second element) can be deleted without violatig the dependency
```

Comment:
also i think deletion allowed should just be a list of activities tbh. a tuple of booleans feels like unnecessary complication